File: ios/core/PlayIntentResolver.swift
import Foundation

enum PlayPauseResolution {
  case playing
  case paused
  case keepCurrent
}

struct PlayIntentResolver {
  private(set) var wantsToPlay = false

  mutating func onPlay() { wantsToPlay = true }
  mutating func onPause() { wantsToPlay = false }
  mutating func onEnded() { wantsToPlay = false }
  mutating func onError() { wantsToPlay = false }
  mutating func onSourceChange() { wantsToPlay = false }
  mutating func onRelease() { wantsToPlay = false }

  func resolve(isPlaying: Bool) -> PlayPauseResolution {
    if isPlaying { return .playing }
    if wantsToPlay { return .keepCurrent }
    return .paused
  }
}

File: android/src/main/java/com/nitroplay/video/core/PlayIntentResolver.kt
package com.nitroplay.video.core

enum class PlayPauseResolution { PLAYING, PAUSED, KEEP_CURRENT }

class PlayIntentResolver {
  var wantsToPlay = false
    private set

  fun onPlay() { wantsToPlay = true }
  fun onPause() { wantsToPlay = false }
  fun onEnded() { wantsToPlay = false }
  fun onError() { wantsToPlay = false }
  fun onSourceChange() { wantsToPlay = false }
  fun onRelease() { wantsToPlay = false }

  fun resolve(isPlaying: Boolean): PlayPauseResolution {
    if (isPlaying) return PlayPauseResolution.PLAYING
    if (wantsToPlay) return PlayPauseResolution.KEEP_CURRENT
    return PlayPauseResolution.PAUSED
  }
}
